lib/vagrant-service-manager/action/setup_network.rb

```diff
@@ -0,0 +1,37 @@
+module Vagrant
+  module ServiceManager
+    module Action
+      class SetupNetwork
+
+        def initialize(app, env)
+          @app = app
+          @machine = env[:machine]
+          @ui = env[:ui]
+        end
+
+        def call(env)
+          add_prerequisite_network if virtualbox? && !private_network_exists?
+          @app.call(env)
+        end
+
+        private
+
+        def virtualbox?
+          @machine.provider.instance_of?(VagrantPlugins::ProviderVirtualBox::Provider)
+        end
+
+        def private_network_exists?
+          @machine.config.vm.networks.any? { |type, _| type == :private_network }
+        end
+
+        def add_prerequisite_network
+          return unless virtualbox? && !private_network_exists?
+
+          @ui.info 'Virtualbox requires an additional private network. Adding it...'
```

Comment:
Is a DHCP interface really what we want? Should it not be the default IP we are using 10.1.2.2.
This should be paired with the ability to change the private network IP via a config option. With this in place we are again on step closer to getting fully out of the box experience for the user.